### PR TITLE
MCO-1314: OCPBUGS-43509: Pick up openshift/kubernetes 1.31 rebase updates

### PR DESCRIPTION
**Pull request by djoshy**
- Bumped dependencies:

```
go get -u k8s.io/api
go get -u k8s.io/code-generator
go get -u k8s.io/component-base
go get -u k8s.io/kube-proxy
go get -u k8s.io/utils
go get -u k8s.io/klog
go get -u k8s.io/klog/v2
go get -u sigs.k8s.io/controller-runtime
go get -u github.com/openshift/build-machinery-go
go get -u sigs.k8s.io/controller-tools
go get -u github.com/openshift/api
go get -u github.com/openshift/client-go
go get -u github.com/openshift/library-go
go get -u k8s.io/cri-api
go get -u k8s.io/kubectl
go get -u k8s.io/kubelet
 
make go-deps
```
- minor changes to how `envtest` and `utilruntime.ErrorHandlers` is invoked
- lint fixes for the rate limited workqueues used in our controllers and how string formats are handled

### CODE REVIEW

File: pkg/controller/build/pod_build_controller.go [hunk 1]
```diff
@@ -63,9 +63,11 @@ func newPodBuildController(
 		Clients:       clients,
 		informers:     newInformers(clients),
 		eventRecorder: eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "machineosbuilder-podbuildcontroller"}),
-		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "machineosbuilder-podbuildcontroller"),
-		config:        ctrlConfig,
-		podHandler:    podHandler,
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
```
Comment (LorbusChris) [inline pkg/controller/build/pod_build_controller.go:66]: nit: indentation

File: pkg/controller/container-runtime-config/container_runtime_config_controller.go [hunk 2]
```diff
@@ -148,8 +148,10 @@ func New(
 		client:        mcfgClient,
 		configClient:  configClient,
 		eventRecorder: ctrlcommon.NamespacedEventRecorder(eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "machineconfigcontroller-containerruntimeconfigcontroller"})),
-		queue:         workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "machineconfigcontroller-containerruntimeconfigcontroller"),
-		imgQueue:      workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter()),
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
```
Comment (LorbusChris) [inline pkg/controller/container-runtime-config/container_runtime_config_controller.go:151]: indentation

File: pkg/controller/drain/drain_controller.go [hunk 3]
```diff
@@ -119,10 +119,12 @@ func New(
...
 
 	ctrl := &Controller{
-		client:               mcfgClient,
-		kubeClient:           kubeClient,
-		eventRecorder:        ctrlcommon.NamespacedEventRecorder(eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "machineconfigcontroller-nodecontroller"})),
-		queue:                workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "machineconfigcontroller-nodecontroller"),
+		client:        mcfgClient,
+		kubeClient:    kubeClient,
+		eventRecorder: ctrlcommon.NamespacedEventRecorder(eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "machineconfigcontroller-nodecontroller"})),
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig(
```
Comment (LorbusChris) [inline pkg/controller/drain/drain_controller.go:125]: indentation

File: pkg/daemon/pinned_image_set.go [hunk 4]
```diff
@@ -155,9 +155,11 @@ func NewPinnedImageSetManager(
...
 		minStorageAvailableBytes: minStorageAvailableBytes,
 		featureGatesAccessor:     featureGatesAccessor,
-		queue:                    workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "pinned-image-set-manager"),
-		prefetchCh:               make(chan prefetch, defaultPrefetchWorkers*2),
-		criClient:                criClient,
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig[string](
+			workqueue.DefaultTypedControllerRateLimiter[string](),
+			workqueue.TypedRateLimitingQueueConfig[string]{Name: "pinned-image-set-manager"}),
+		prefetchCh: make(chan prefetch, defaultPrefetchWorkers*2),
+		criClient:  criClient,
```
Comment (LorbusChris) [inline pkg/daemon/pinned_image_set.go:162]: indentation

File: pkg/controller/build/build_controller.go [hunk 5]
```diff
@@ -210,8 +210,10 @@ func newBuildController(
 		informers:     newInformers(clients),
 		Clients:       clients,
 		eventRecorder: eventBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: "machineosbuilder-buildcontroller"}),
-		mosQueue:      workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "machineosbuilder"),
-		config:        ctrlConfig,
+		mosQueue: workqueue.NewTypedRateLimitingQueueWithConfig(
+			workqueue.DefaultTypedControllerRateLimiter[string](),
+			workqueue.TypedRateLimitingQueueConfig[string]{Name: "machineosbuilder"}),
+		config: ctrlConfig,
```
Comment (LorbusChris) [inline pkg/controller/build/build_controller.go:216]: nit: indentation

**Review by LorbusChris — APPROVED** [review]
There's a couple of second-column indentations that don't match their surroundings in the commit `lint fixes for kube rebase v1.31.1`, but I wouldn't personally block on that.

/lgtm
/retest-required

**Comment by djoshy** [comment]
/hold

Thanks @LorbusChris! Holding for QE review

**Comment by sergiordlr** [comment]
A test run with the critical e2e test cases has been launched  https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-aws-ipi-longduration-mco-critical-f9/1849365598086107136

**Comment by sdodson** [comment]
I just wanted https://github.com/openshift/machine-config-operator/pull/4629/commits/b53b9df3b05a4d3fc0cc6a43bd0b7cccd1d757b6 I think

**Pull request merged**